### ISSUE TITLE
Давать содержимому модулей модификатор internal

### ISSUE THREAD

**Issue by SunSerega** [pascalabcnet/pascalabcnet#2302]
Связанное: #2018, http://forum.mmcs.sfedu.ru/t/pascalabc-net/2530/351?u=sun_serega

Содержимое модулей (типы, подпрограммы, глобальные переменные) должны получать модификатор `internal` а не `public`.
Иначе в пользовательские .dll попадают кишки модулей (а точнее их огрызки), которые не нужны тому, кто подключает эту .dll .

Но - осторожно, содержимое неймспейсов, основного файла библиотеки (который начинается со слова `library`) и генерируемый класс, содержащий подпрограмму `void Main()` - должны остаться с модификатором `public`.

**Comment by ibond84** [comment]
Нет, там куча проблем возникает. Например если есть в основном пространстве имен публичное свойство с типом из другого модуля, тип не может быть internal.

**Comment by ibond84** [comment]
Кстати альтернатива import, правда работает только в паскале..

```
library lib1;

uses unit1;

type t1 = unit1.t1;
```

**Comment by ibond84** [comment]
```
type 
  [Import]
  t1 = unit1.t1;
```
А такое бы переносило бы тип в основное пространство имен (для C#)

**Comment by SunSerega** [comment]
> Например если есть в основном пространстве имен публичное свойство с типом из другого модуля, тип не может быть internal.

Не может быть потому что библиотека не соберётся, или потому что этому свойству не получится присвоить новый экземпляр?

Во втором случае это вообще не проблема. Свойство можно использовать не создавая экземпляр. А когда экземпляры надо создавать - это значит что надо таки сделать синоним типа в главном файле библиотеки (или в 1 из пространств имён).

**Comment by ibond84** [comment]
Не может, потому что рантайм это запрещает. При использовании свойства будет ошибка времени выполнения

**Comment by AIexandrKotov** [comment]
А может лучше сделать модификаторы доступа для типов. чтобы пользователь сам выбирал что public, а что internal? Issue же давно висит.
